### PR TITLE
ci: group check steps into an individual queue

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

https://discord.com/channels/428295358100013066/560503042458517505/1074785045107196066

#### Summary of Changes

made these steps run on bk agent which queue is `check` (has already set up them on bk)

- sanity (about 15s)
- shellcheck (about 19 s)
- checks (about 5~7 mins)

### CODE REVIEW

File: ci/buildkite-pipeline.sh [hunk 1]
```diff
@@ -139,7 +139,14 @@ wait_step() {
 }
 
 all_test_steps() {
-  command_step checks ". ci/rust-version.sh; ci/docker-run.sh \$\$rust_nightly_docker_image ci/test-checks.sh" 20
```
Comment (t-nelson) [inline ci/buildkite-pipeline.sh:142]: might be a little cleaner to add queue a fourth optional arg to `command_step_checks`? something like

```
local queue="${4:-solana}"
```

and replace the hardcoded queue accordingly

Comment (yihau) [reply ci/buildkite-pipeline.sh:142]: fixed 🪖

File: ci/buildkite-pipeline.sh [hunk 2]
```diff
@@ -114,7 +114,7 @@ command_step() {
     timeout_in_minutes: $3
     artifact_paths: "log-*.txt"
     agents:
-      queue: "solana"
+      queue: ${4:-solana}
```
Comment (t-nelson) [inline ci/buildkite-pipeline.sh:117]: is this still valid TOML if we drop the quotes?

Comment (yihau) [reply ci/buildkite-pipeline.sh:117]: I think it is yaml 🤔 btw, I found we have quotes and no quotes in the config. I will create a PR to make them all have quotes

<details>
    <summary>steps</summary>

```yaml

steps:

  - name: "sanity"

    command: "ci/test-sanity.sh"

    timeout_in_minutes: 5

    artifact_paths: "log-*.txt"

    agents:

      queue: check

  - wait

  - name: "shellcheck"

    command: "ci/shellcheck.sh"

    timeout_in_minutes: 5

    artifact_paths: "log-*.txt"

    agents:

      queue: check

  - wait

  - name: "checks"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_nightly_docker_image ci/test-checks.sh"

    timeout_in_minutes: 20

    artifact_paths: "log-*.txt"

    agents:

      queue: check

  - wait

  - name: "coverage"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_nightly_docker_image ci/test-coverage.sh"

    timeout_in_minutes: 80

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - wait

  - name: "stable"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_stable_docker_image ci/test-stable.sh"

    timeout_in_minutes: 70

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - name: "doctest"

    command: "ci/test-docs.sh"

    timeout_in_minutes: 15

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - wait

  - command: "ci/test-stable-sbf.sh"

    name: "stable-sbf"

    timeout_in_minutes: 35

    artifact_paths: "sbf-dumps.tar.bz2"

    agents:

      queue: "solana"

  - command: "ci/test-stable-perf.sh"

    name: "stable-perf"

    timeout_in_minutes: 35

    artifact_paths: "log-*.txt"

    agents:

      queue: "cuda"

  - command: "scripts/build-downstream-projects.sh"

    name: "downstream-projects"

    timeout_in_minutes: 35

    agents:

      queue: "solana"

  - name: "wasm"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_stable_docker_image ci/test-wasm.sh"

    timeout_in_minutes: 20

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - name: "bench"

    command: "ci/test-bench.sh"

    timeout_in_minutes: 40

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - name: "local-cluster"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_stable_docker_image ci/test-local-cluster.sh"

    timeout_in_minutes: 40

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - name: "local-cluster-flakey"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_stable_docker_image ci/test-local-cluster-flakey.sh"

    timeout_in_minutes: 10

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - name: "local-cluster-slow-1"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_stable_docker_image ci/test-local-cluster-slow-1.sh"

    timeout_in_minutes: 40

    artifact_paths: "log-*.txt"

    agents:

      queue: solana

  - name: "local-cluster-slow-2"

    command: ". ci/rust-version.sh; ci/docker-run.sh $$rust_stable_docker_image ci/test-local-cluster-slow-2.sh"

    timeout_in_minutes: 40

    artifact_paths: "log-*.txt"

    agents:

      queue: solana
```

</details>

Comment (yihau) [reply ci/buildkite-pipeline.sh]: we don't need the PR anymore cuz the no quote string is also from here. when I added quote, the problem is solved 🤞

Comment (yihau) [reply ci/buildkite-pipeline.sh]: nvm. I think I need more coffee to reboot myself. 😴 

I added the quotes for not changing the behavior

Comment (t-nelson) [reply ci/buildkite-pipeline.sh]: thanks for investigating!

**Review by t-nelson — APPROVED** [review]
lgtm, thanks!

**Pull request merged**